### PR TITLE
🏗 Limit commit log size in Travis logs and print branch point info

### PR DESCRIPTION
**Pull request by rsimha**
This PR avoids extremely long change summaries in Travis logs, when something is wrong with the way a branch was forked. See https://travis-ci.org/ampproject/amphtml/jobs/470147300#L616 for example.

It also prints the branch point in the Travis logs.

I suspect the occasional extra-long commit logs on Travis (computed using `TRAVIS_COMMIT_RANGE`) might be related to https://github.com/travis-ci/travis-ci/issues/4596. This could also happen when a branch on a fork was forked from some other branch, that doesn't track `origin/master`. Either way, we should print a reasonable number of commits in the change summary.

Follow up to #19954

### CODE REVIEW

**Comment by rsimha** [comment]
/to @danielrozenberg 

https://en.wikipedia.org/wiki/The_Saga_Continues_(Wu-Tang_Clan_album)

File: build-system/git.js [hunk 1]
```diff
@@ -60,17 +60,23 @@ exports.gitDiffStatMaster = function() {
...
 /**
  * Returns a detailed log of commits included in a PR check, starting with (and
- * including) the branch point off of master.
+ * including) the branch point off of master. Limited to at most 100 commits to
+ * keep the output sane.
  *
  * @return {string}
  */
 exports.gitDiffCommitLog = function() {
+  const maxCount = 100;
```
Comment (danielrozenberg) [inline build-system/git.js]: Nit: rather have this at the top of the file

File: build-system/git.js [hunk 2]
```diff
@@ -60,17 +60,23 @@ exports.gitDiffStatMaster = function() {
...
     exports.gitPrBranchPoint() : exports.gitBranchPointFromMaster();
-  return getStdout(`git -c color.ui=always log --graph --pretty=format:\
-"%C(red)%h%C(reset) %C(bold cyan)%an%C(reset) -%C(yellow)%d%C(reset) \
-%C(reset)%s%C(reset) %C(green)(%cr)%C(reset)" \
---abbrev-commit ${branchPoint}^...HEAD`).trim();
+  let commitLog = getStdout(`git -c color.ui=always log --graph \
+--pretty=format:"%C(red)%h%C(reset) %C(bold cyan)%an%C(reset) \
+-%C(yellow)%d%C(reset) %C(reset)%s%C(reset) %C(green)(%cr)%C(reset)" \
+--abbrev-commit ${branchPoint}^...HEAD --max-count=${maxCount}`).trim();
+  if (commitLog.split('\n').length >= maxCount) {
```
Comment (danielrozenberg) [inline build-system/git.js]: Do you count the boilerplate/prefix lines here?

File: build-system/pr-check.js [hunk 3]
```diff
@@ -102,9 +104,11 @@ function printChangeSummary() {
       colors.cyan(process.env.TRAVIS_PULL_REQUEST_SHA));
   console.log(filesChanged);
 
+  const branchPoint = process.env.TRAVIS ?
+    gitPrBranchPoint() : gitBranchPointFromMaster();
```
Comment (danielrozenberg) [inline build-system/pr-check.js]: This change seems unrelated. Need to update the description?

File: build-system/git.js (hunk 2 above)
Comment (rsimha) [reply build-system/git.js]: This didn't work, so I removed it.

File: build-system/pr-check.js (hunk 3 above)
Comment (rsimha) [reply build-system/pr-check.js]: It's part of the commit log logic. Will update.

File: build-system/git.js (hunk 1 above)
Comment (rsimha) [reply build-system/git.js]: Done.

**Comment by rsimha** [comment]
@danielrozenberg PTAL

**Comment by danielrozenberg** [comment]
I have an alternate theory - notice that Travis does `git clone ... --max_depth = 50` - so if the branch-off point is older than 50 commits ago (which is very likely in our project!) it might not have information about that commit. Maybe. :smile:
